### ISSUE TITLE
`oras manifest delete` failed to delete last referrers in registry without referrer API support

### ISSUE THREAD

**Issue by qweeah** [oras-project/oras#1112]
### What happened in your environment?

Tried to delete the only referrer of an artifact in OCI-1.0 compliant registry without referrers API support but failed.

### What did you expect to happen?

The referrer should be able to be deleted from the referrers index.

### How can we reproduce it?

```console
# 1. localhost:6000 is an OCI 1.0 compliant registry
oras discover localhost:6000/test:foobar -o tree
localhost:6000/test@sha256:fd6ed2f36b5465244d5dc86cb4e7df0ab8a9d24adc57825099f522fe009a22bb
└── test/sbom.file
    └── sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d
        └── test/signature.file
            └── sha256:0cb8c4da7e9ff2e7eefca33141091b9239218e3125a35e17e8bcd05fa3a5e714

# 2. delete referrers sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d
✗ oras manifest delete -v localhost:6000/test@sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d
Are you sure you want to delete the manifest "sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d" and all tags associated with it? [y/N] y
Deleted localhost:6000/test@sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d

# 3. manifest doesn't exist any more
✗ oras manifest fetch localhost:6000/test@sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d
Error: failed to fetch the content of "localhost:6000/test@sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d": localhost:6000/test@sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d: not found

# 4. but tag of sha256-fd6ed2f36b5465244d5dc86cb4e7df0ab8a9d24adc57825099f522fe009a22bb still returns the deleted referrer
oras discover localhost:6000/test:foobar -o tree
localhost:6000/test@sha256:fd6ed2f36b5465244d5dc86cb4e7df0ab8a9d24adc57825099f522fe009a22bb
└── test/sbom.file
    └── sha256:e2c6633a79985906f1ed55c592718c73c41e809fb9818de232a635904a74d48d
        └── test/signature.file
            └── sha256:0cb8c4da7e9ff2e7eefca33141091b9239218e3125a35e17e8bcd05fa3a5e714
```

### What is the version of your ORAS CLI?

1.1.0

### What is your OS environment?

Ubuntu 20.04

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

**Comment by qweeah** [comment]
This relies on the fix of https://github.com/oras-project/oras-go/issues/592

**Comment by qweeah** [comment]
Resolved by #1154
